internal/incident/incident.go

```diff
@@ -474,8 +470,10 @@ func (i *Incident) evaluateEscalations(eventTime time.Time) ([]*rule.Escalation,
 
 	for rID := range i.Rules {
 		r := i.runtimeConfig.Rules[rID]
-
-		if r == nil || !r.IsActive.Valid || !r.IsActive.Bool {
+		if r == nil {
+			i.logger.Warnw("Incident refers unknown rule",
+				zap.String("incident", i.String()),
```

Comment:
You don't have to add `incident` here, as `i.logger` should already contain that key.

Reply:
I'm honestly unsure. It seems like `i.logger` only gets populated with `zap.String("incident", i.String())` during `ProcessEvent` if this incident is seen as "new", defined by having a zero `started_at` value.

https://github.com/Icinga/icinga-notifications/blob/853992e30328b3292dbcf018df95fe9bbcbe3a31/internal/incident/incident.go#L149-L157

I mean yes, I guess it can be removed at this position. But does it might occur that some Incidents are without this field, e.g., for reevaluation?

Reply:
That sounds like if there are cases where the incident logger doesn't contain metadata fields for the incident, that's a bug.

Reply:
> But does it might occur that some Incidents are without this field, e.g., for reevaluation?

Nope!
https://github.com/Icinga/icinga-notifications/blob/e954e4059911d3bd9b6d6720a4de13f65609f2e8/internal/incident/incidents.go#L124-L125

Reply:
Thanks @yhabteab. I somehow missed that one.